catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/QuerySources.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package org.codice.alliance.nsili.endpoint;
+
+import java.util.Set;
+
+/**
+ * Accesses and modifies {@link org.codice.alliance.nsili.endpoint.NsiliEndpoint#querySources}. Used
+ * by {@link org.codice.alliance.catalog.nsili.sourcestoquery.ui.service.SourcesToQuery}.
+ */
+public interface QuerySources {
```

Comment:
write some javadoc for each method, can methods return null? can method parameters be null?